test/e2e/setup.go

```diff
@@ -167,7 +167,8 @@ func setup(ctx context.Context) error {
 }
 
 func done(ctx context.Context) error {
-	if os.Getenv("E2E_DELETE_CLUSTER") != "" {
+	// terminate early if delete flag is set to false
+	if os.Getenv("CI") != "" || os.Getenv("E2E_DELETE_CLUSTER") != "" {
```

Comment:
so `os.Getenv("E2E_DELETE_CLUSTER") != ""` here is the legacy case which we'll get rid of, but you're missing the new case which is `if os.Getenv("E2E_DELETE_CLUSTER") == "false" { return nil }` ?